pkg/devfile/detect.go

```diff
@@ -233,24 +233,33 @@ func (a AlizerClient) Analyze(path string) ([]language.Language, error) {
 
 // SelectDevFileFromTypes is a wrapper call to Alizer's SelectDevFileFromTypes()
 func (a AlizerClient) SelectDevFileFromTypes(path string, devFileTypes []recognizer.DevFileType) (recognizer.DevFileType, error) {
-	return recognizer.SelectDevFileFromTypes(path, devFileTypes)
+	index, err := recognizer.SelectDevFileFromTypes(path, devFileTypes)
+	return devFileTypes[index], err
+}
+
+func (a AlizerClient) DetectComponents(path string) ([]recognizer.Component, error) {
+	return recognizer.DetectComponents(path)
 }
 
 // AnalyzeAndDetectDevfile analyzes and attempts to detect a devfile from the devfile registry for a given local path
 func AnalyzeAndDetectDevfile(a Alizer, path, devfileRegistryURL string) ([]byte, string, string, error) {
 	var devfileBytes []byte
-
-	alizerLanguages, err := a.Analyze(path)
+	alizerDevfileTypes, err := getAlizerDevfileTypes(devfileRegistryURL)
 	if err != nil {
 		return nil, "", "", err
 	}
 
-	alizerDevfileTypes, err := getAlizerDevfileTypes(devfileRegistryURL)
+	alizerComponents, err := a.DetectComponents(path)
 	if err != nil {
 		return nil, "", "", err
 	}
 
-	for _, language := range alizerLanguages {
+	if len(alizerComponents) == 0 {
+		return nil, "", "", &NoDevfileFound{Location: path}
+	}
+
+	// Assuming it's a single component. as multi-component should be handled before
```

Comment:
so what does this mean? `DetectComponents` can return multple components and each component have different languages?

Reply:
Because previously AlizerLanguage was an array []Language{} but now AlizerComponent is an array where Language is inside - []Component{{path, []Language}}

so what are the possible values for AlizerComponent, can you give an example? Since we are taking the index 0 always

Reply:
for example, running `DetectComponent` against a single component project, the output will be:
```
[
 {
   path: /Users/stephanie/Documents/devfile-samples/devfile-sample-dotnet-basic/ 
   languages: [{C# [csharp dotnet] 15 [net6.0] [] true} {JavaScript [js node nodejs] 75 [] [] true} {Smalltalk [squeak] 15 [] [] false} {GCC Machine Description [] 10 [] [] false}] 
  }
 ]
```

Reply:
The output for languages have no differences here.

Reply:
we have discussed in the meeting. currently we have multi-component field in the spec, and after [the issue ](https://issues.redhat.com/projects/DEVHAS/issues/DEVHAS-80) is in, We will detect if the project is a multi component project when we first scanning through the project. 
so when come to the analyze & look for devfile stage, we always assume it's a single component .